docker/src/main/java/brooklyn/entity/container/docker/DockerInfrastructureImpl.java

```diff
@@ -101,6 +102,8 @@
 
     private transient Object mutex = new Object[0];
 
+    private static final AtomicInteger infrastructureID = new AtomicInteger();
```

Comment:
Forgot, we probably also want to stuff this into an `AttributeSensor<AtomicInteger>` for persistence.